goaviatrix/tags.go

```diff
@@ -50,81 +48,81 @@ func (c *Client) AddTags(tags *Tags) error {
 }
 
 func (c *Client) GetTags(tags *Tags) ([]string, error) {
-	tags.CID = c.CID
-	tags.Action = "list_resource_tags"
-	resp, err := c.Post(c.baseURL, tags)
-	if err != nil {
-		return nil, errors.New("HTTP Post list_resource_tags failed: " + err.Error())
-	}
-	var data TagAPIResp
-	buf := new(bytes.Buffer)
-	buf.ReadFrom(resp.Body)
-	bodyString := buf.String()
-	bodyIoCopy := strings.NewReader(bodyString)
-	if err = json.NewDecoder(bodyIoCopy).Decode(&data); err != nil {
-		return nil, errors.New("Json Decode list_resource_tags failed: " + err.Error() + "\n Body: " + bodyString)
+	data := map[string]string{
+		"action":        "list_resource_tags",
+		"CID":           c.CID,
+		"cloud_type":    strconv.Itoa(tags.CloudType),
+		"resource_type": tags.ResourceType,
+		"resource_name": tags.ResourceName,
 	}
-	if !data.Return {
-		return nil, errors.New("Rest API list_resource_tags Post failed: " + data.Reason)
+	var resp TagAPIResp
+	err := c.GetAPI(&resp, data["Action"], data, BasicCheck)
+	if err != nil {
+		return nil, fmt.Errorf("HTTP Get list_resource_tags failed: %v", err)
```

Comment:
You can just return the naked err from GetAPI since it already nicely formats the error